godal.go

```diff
@@ -2861,6 +2861,7 @@ func (f *Feature) SetGeometry(geom *Geometry, opts ...SetGeometryOption) error {
 	return cgc.close()
 }
 
+// Deprecated: when running with GDAL 3.6+, you may use SetGeometryColumnName on Layer
```

Comment:
I'm thinking we should change this method's signature to return an error, and explicitely return a go error without going into cgo/gdal for version>=3.6

Reply:
or for version>3.9 actually no?

Reply:
Comments and method signature changed. Tell me if it's ok so I squash commits.

Reply:
When we look at the coverage, I wonder if we may create a cpp function so we check version in cpp code, like for other functions.

Reply:
As you prefer. I am not very concerned by the coverage percent for this particular issue. If you were to move it all to cpp (which is effectively cleaner and a bit more in line with what we do elsewhere in godal), your coverage would still not be ideal as it would not be possible to write a test handling both the error and no-error cases.

Reply:
New changes done. When moving to cpp, as errors are handled in a proper way, we don't even need to check version.